frontend/src/app/services.ts

```diff
@@ -91,14 +91,34 @@ export const hitasApi = createApi({
     baseQuery: fetchBaseQuery({
         baseUrl: Config.api_url,
         prepareHeaders: (headers) => {
-            headers.set("Authorization", "Bearer " + Config.token);
+            Config.token && headers.set("Authorization", "Bearer " + Config.token);
             return headers;
         },
+        ...(!Config.token && {credentials: "include"}),
     }),
     tagTypes: ["HousingCompany", "Apartment", "Index", "Owner"],
     endpoints: (builder) => ({}),
 });
 
+export const authApi = createApi({
+    reducerPath: "authApi",
+    baseQuery: fetchBaseQuery({
+        baseUrl: Config.auth_api_url,
+        credentials: "include",
+    }),
+    endpoints: (builder) => ({}),
+});
+
+const userApi = authApi.injectEndpoints({
+    endpoints: (builder) => ({
+        getUserInfo: builder.query<IUserInfoResponse, void>({
+            query: () => ({
+                url: "userinfo",
+            }),
+        }),
+    }),
+});
```

Comment:
Right now the userInfo is not cached at all, which means that on every page change the frontend queries the API to get the userInfo. We can cache the request's results with tags like so:

```ts
export const authApi = createApi({
    ...
    tagTypes: ["UserInfo"],
    endpoints: (builder) => ({
        getUserInfo: builder.query<IUserInfoResponse, void>({
            query: () => ({
                url: "userinfo",
            }),
            providesTags: ["UserInfo"],
        }),
    }),
});
```

---

After we have cached the data and as the userInfo pretty much never changes, we can extend the default cache duration with `keepUnusedDataFor` so we should need to fetch the userInfo only once per authentication
https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#reducing-subscription-time-with-keepunuseddatafor

---

After we have the cache enabled, you should then invalidate the cache when the user logs out, so that the userInfo is not shown  after user is already logged out.
https://redux-toolkit.js.org/rtk-query/api/created-api/api-slice-utils#resetapistate

Reply:
I tested this with the tags and keepUnusedDataFor but these do not fix the problem. It should cache the data by default without tags and keep the cache on App component, which is not unmounted. I don't know why it still seems to call the endpoint on some route changes instead of using the cache.

Reply:
👍 Not a major issue, we can resolve that one later if its needed